x/programs/cmd/simulator/src/lib.rs

```diff
@@ -198,7 +170,8 @@ pub struct PlanResult {
     /// The timestamp of the function call response.
     pub timestamp: u64,
     /// The result of the function call.
-    pub response: Option<String>,
+    #[serde_as(as = "Base64")]
+    pub response: Vec<u8>,
```

Comment:
Yeah, after looking at the examples, I think these types are super weird now. `PlanResult` should be `StepResult<T>` (rename and make it generic) where the response is `Result<T, String>`.

Reply:
Good idea!

Reply:
thanks